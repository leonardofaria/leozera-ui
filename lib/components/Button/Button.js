/*
 * Button component
 *
 * @usage
 * <Button>Hello Stranger</Button>
 */
import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames/bind'

import styles from './Button.scss'

const cx = classNames.bind(styles) // eslint-disable-line

function Button(props) {
  const {
    href, children, onClick, disabled, className, ...attrs
  } = props
  const Element = href ? 'a' : 'button'
  const role = Element === 'a' ? 'button' : null
  const btnClass = cx('btn', className)

  return (
    <Element
      className={btnClass}
      href={href}
      onClick={onClick}
      role={role}
      disabled={disabled}
      {...attrs}
    >
      {children}
    </Element>
  )
}

Button.propTypes = {
  /** Pass an href prop to make the button an `a` element instead of a `button` */
  href: PropTypes.string,
  /** Content for the button */
  children: PropTypes.node.isRequired,
  /** Function to run when the button is clicked */
  onClick: PropTypes.func,
  /** Disable button */
  disabled: PropTypes.bool,
  /** CSS classes */
  className: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.string,
  ]),
}

Button.defaultProps = {
  href: null,
  onClick: null,
  disabled: false,
  className: null,
}

export default Button
