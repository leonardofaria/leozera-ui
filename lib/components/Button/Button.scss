@import '../../styles/base.scss';

.btn {
  border: 0;
  color: $primary-gray;
}
