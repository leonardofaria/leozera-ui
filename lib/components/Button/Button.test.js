import React from 'react';
import { shallow } from 'enzyme';

import Button from './Button';

describe('Tests for <Button /> Component', () => {
  describe('Initial rendering', () => {
    it('should render a proper component', () => {
      const wrapper = shallow(<Button>Hello Stranger</Button>);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render a <Button> with the default class', () => {
      const wrapper = shallow(<Button>Hello Stranger</Button>);
      expect(wrapper.hasClass('btn')).toBeTruthy();
    });

    it('should render a <Button> with a custom class', () => {
      const wrapper = shallow(
        <Button className="btn-test">Hello Stranger</Button>,
      );
      expect(wrapper.hasClass('btn-test')).toBeTruthy();
    });
  });
});
