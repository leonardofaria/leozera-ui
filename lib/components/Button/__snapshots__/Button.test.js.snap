// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Button /> Component Initial rendering should render a proper component 1`] = `ShallowWrapper {}`;
